message.go

```diff
@@ -13,7 +15,31 @@ type rpcMessage interface {
 	DecodeMessage(int, decoder, *protocolHandler, *callContainer) error
 }
 
+type BasicRPCData struct {
+	ctx context.Context
+}
+
+func (r *BasicRPCData) Context() context.Context {
+	if r.ctx == nil {
+		r.ctx = context.Background()
```

Comment:
Or you could just return `context.Background()` without setting `r.ctx` -- that works, too.